oak-store-composite/src/main/java/org/apache/jackrabbit/oak/composite/CompositionContext.java

```diff
@@ -67,10 +69,10 @@ class CompositionContext {
         this.prefetchNodeStore = globalStore instanceof PrefetchNodeStore ? (PrefetchNodeStore) globalStore : PrefetchNodeStore.NOOP;
         this.nonDefaultStores = nonDefaultStores;
 
-        ImmutableSet.Builder<MountedNodeStore> b = ImmutableSet.builder();
+        Set<MountedNodeStore> b = new LinkedHashSet<>();
         b.add(this.globalStore);
         b.addAll(this.nonDefaultStores);
```

Comment:
Both values `globalStore` & `nonDefaultStores` can't be null else the code would have already failed in the higher tier.